### PR TITLE
Allow chat clients to connect to non-localhost servers

### PR DESCRIPTION
**Pull request by alexras**

### CODE REVIEW

File: hydroflow/examples/chat/client.rs [hunk 1]
```diff
@@ -13,9 +13,19 @@ use tokio_stream::wrappers::LinesStream;
 
 pub(crate) async fn run_client(opts: Opts) {
     // set up network and I/O channels
-    let server_addr: SocketAddr = format!("{}:{}", opts.addr, opts.port).parse().unwrap();
+    let server_ip = match opts.server_addr {
+        Some(ip) => ip,
+        None => panic!("Clients must specify --server-addr"),
+    };
```
Comment (MingweiSamuel) [inline hydroflow/examples/chat/client.rs:19]: Rust: simplify this to `let server_ip = opts.server_addr.expect("Clients must specify --server-addr")`

File: hydroflow/examples/chat/client.rs [hunk 2]
```diff
@@ -13,9 +13,19 @@ use tokio_stream::wrappers::LinesStream;
...
+
+    let server_port = match opts.server_port {
+        Some(port) => port,
+        None => panic!("Clients must specify --server-port"),
+    };
```
Comment (MingweiSamuel) [inline hydroflow/examples/chat/client.rs:24]: Rust: simplify this to `let server_port = opts.server_port.expect("Clients must specify --server-port");`

**Review by MingweiSamuel — APPROVED** [review]
Use `.expect()` for the options and run `cargo fmt` then merge after CI passes!

File: hydroflow/examples/chat/client.rs (hunk 1 above)
Comment (alexras) [reply hydroflow/examples/chat/client.rs:19]: Nice, thank you!

**Pull request merged**